src/main/java/de/tum/in/www1/artemis/service/exam/ExamRegistrationService.java

```diff
@@ -127,6 +137,7 @@ public List<ExamUserDTO> registerStudentsForExam(Long courseId, Long examId, Lis
                     exam.addExamUser(examUser);
                 }
 
+                addStudentToExamChannel(exam, student);
```

Comment:
This method call the DB each time to get the exam channel, doesn't it?
So if I add 2000 students with this method, will this crease 2000 DB calls? 🤔 

Since you already have `ChannelService#deregisterUsersFromExamChannel`, I suggest to create a similar method to add register students to the exam channel as well.

Reply:
Agree, please make sure that adding to the channel is done asynchronously

Reply:
We created a new method `ChannelService#registerUsersToExamChannel` and add students to the exam channel asynchronously.